rewrite-core/src/main/java/org/openrewrite/text/CreateTextFile.java

```diff
@@ -92,13 +95,17 @@ public Collection<SourceFile> generate(AtomicBoolean shouldCreate, ExecutionCont
     @Override
     public TreeVisitor<?, ExecutionContext> getVisitor(AtomicBoolean created) {
         Path path = Paths.get(relativeFileName);
-        return new PlainTextVisitor<ExecutionContext>() {
+        return new TreeVisitor<SourceFile, ExecutionContext>() {
             @Override
-            public PlainText visitText(PlainText text, ExecutionContext ctx) {
-                if ((created.get() || Boolean.TRUE.equals(overwriteExisting)) && path.toString().equals(text.getSourcePath().toString())) {
-                    return text.withText(fileContents);
+            public @Nullable SourceFile visit(@Nullable Tree tree, ExecutionContext executionContext) {
```

Comment:
AFAICT the `@Nullable` annotation isn't required here.

Reply:
I've removed the first one on the return type; the second on on the first visit argument seems common throughout our code base.